### PR TITLE
Create MultiFunctionTemplateView and MultiFunctionTemplatePresenter

### LINKED ISSUE #36928: Create `MultiFunctionTemplate` presenter and view
**Description**
This issue is to create a `MultiFunctionTemplateView` and `MultiFunctionTemplatePresenter` class to replace the `ConvFunctionTemplateView` and `ConvTemplatePresenter` classes. These new classes should be a more generalised implementation of the old classes, containing very little reference to ConvFit.

When we have achieved this, we can then use the `MultiFunctionTemplate` classes to replace the `IqtTemplateBrowser`, `IqtTemplatePresenter` and `IqtFunctionModel` classes. This will mean these two tabs have an almost identical implementation, and therefore maintenance and extention in the future will be far easier. Currently it is difficult to maintain and extend these tabs.

**Acceptance criteria**
- Classes called `MultiFunctionTemplateView` and `MultiFunctionTemplatePresenter` exist
- These new classes can be used in place of the current ConvFit classes

### PR DESCRIPTION
**Pull request by robertapplin**
### Description of work
This PR creates two generic classes, `MultiFunctionTemplateView` and `MultiFunctionTemplatePresenter`, that can replace the `ConvFunctionTemplateView` and `ConvTemplatePresenter`.

The future plan is to use these two classes for the IqtTab instead of `IqtFunctionTemplateView` and `IqtTemplatePresenter`. This will make it easier to change the StretchedExp checkbox in this tab to a combobox.

Before I can do this, I need to create a `MultiFunctionTemplateModel` as outlined in this next issue #36964

Fixes #36928

### To test:
Code review

Follow the testing instructions here
https://developer.mantidproject.org/Testing/Inelastic/DataAnalysisTests.html

*This does not require release notes* because **there are no functional changes**

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

File: qt/scientific_interfaces/Inelastic/Analysis/FunctionBrowser/MultiFunctionTemplateView.cpp [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    setBoolSilent(m_subTypeProperties[subTypeIndex], value);
+  } else {
+    setEnumSilent(m_subTypeProperties[subTypeIndex], value);
+  }
+}
+
+void MultiFunctionTemplateView::setSubTypes(std::map<std::size_t, int> const &subTypes) {
+  for (auto it = subTypes.cbegin(); it != subTypes.cend(); ++it) {
+    setSubType(it->first, it->second);
+  }
```
Comment (cailafinn) [inline qt/scientific_interfaces/Inelastic/Analysis/FunctionBrowser/MultiFunctionTemplateView.cpp:113]: Could this just be a `std::for_each`?

**Review by cailafinn — CHANGES_REQUESTED** [review]
Just a couple of code comments/questions. Functionally works really well.

File: qt/scientific_interfaces/Inelastic/Analysis/FunctionBrowser/IqtFunctionModel.cpp [hunk 2]
```diff
@@ -276,6 +276,12 @@ std::string IqtFunctionModel::setBackgroundA0(double value) {
   return "";
 }
 
+void IqtFunctionModel::setResolution(const std::vector<std::pair<std::string, size_t>> &fitResolutions) {
+  (void)fitResolutions;
+}
```
[the remark this answers is not in the document]
Comment (robertapplin) [reply qt/scientific_interfaces/Inelastic/Analysis/FunctionBrowser/IqtFunctionModel.cpp:281]: Yes, this is to override an existing pure virtual function. This is likely to change in the next upcoming PR which will refactor a lot of the model code

**Review by cailafinn — APPROVED** [review]
Found an issue while testing but it's not related to this, so I'm happy to approve. 

Unrelated (it's on main) crash when loading invalid data:
1. Produce an Iqt workspace from `Data Manipulation` but use the `irs26176_graphite002_red` file in both boxes
2. Load the Iqt workspace into the `Data Analysis` GUI
3. Change the number of exponentials to 1
4. Crash.

**Comment by robertapplin** [comment]
> Found an issue while testing but it's not related to this, so I'm happy to approve.
> 
> Unrelated (it's on main) crash when loading invalid data:
> 
> 1. Produce an Iqt workspace from `Data Manipulation` but use the `irs26176_graphite002_red` file in both boxes
> 2. Load the Iqt workspace into the `Data Analysis` GUI
> 3. Change the number of exponentials to 1
> 4. Crash.

Thanks for finding this! I have opened an issue here #37004

**Pull request merged**